src/main/java/net/pms/encoders/MEncoderVideo.java

```diff
@@ -500,7 +500,7 @@ public ProcessWrapper launchTranscode(
 		} else if (isAviSynthEngine()) {
 			deferToTsmuxer = false;
 			LOGGER.trace(prependTraceReason + "we are using AviSynth.");
-		} else if (defaultVideoTrack.isH264() && renderer.isH264Level41Limited() && !isVideoWithinH264LevelLimits(defaultVideoTrack, renderer)) {
+		} else if (defaultVideoTrack.isH264() && renderer.getH264LevelLimit() < 4.2 && !isVideoWithinH264LevelLimits(defaultVideoTrack, renderer)) {
```

Comment:
I think now we can make this check more specific by modifying `isVideoWithinH264LevelLimits` to work with the new logic, then change this line to:
```suggestion
		} else if (defaultVideoTrack.isH264() && !isVideoWithinH264LevelLimits(defaultVideoTrack, renderer)) {
```